### PR TITLE
FEAURE: RAIL 949 integrate new table component to AD

### PR DESCRIPTION
**Pull request by tomasreichmann**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1793/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1280/ : FAILURE in 11m 46s

**Comment by tomasreichmann** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-610.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1796/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/397/ : FAILURE in 24m 03s

**Comment by tomasreichmann** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-610.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1801/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1286/ : SUCCESS in 2m 33s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1802/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/398/ : SUCCESS in 24m 46s

File: examples/src/components/PivotTableDynamicExample.jsx [hunk 1]
```diff
@@ -181,6 +184,34 @@ const totalPresets = {
         totalItem: createColumnTotal(franchiseFeesAdRoyaltyIdentifier, locationStateDisplayFormIdentifier, 'max')
     }
 };
+const filterPresets = {
+    attributeCalifornia: {
+        label: 'Attribute (California)',
+        key: 'attributeCalifornia',
+        filterItem: createPositiveAttributeFilter(
+            locationStateDisplayFormIdentifier,
+            [locationStateAttributeCaliforniaUri]
```
Comment (lhorak) [inline examples/src/components/PivotTableDynamicExample.jsx]: Is this supposed to be in array? Just checking :)

Comment (lhorak) [inline examples/src/components/PivotTableDynamicExample.jsx]: Is this supposed to be in array? Just checking :)

Comment (lhorak) [reply examples/src/components/PivotTableDynamicExample.jsx]: found it 👍

File: src/components/core/base/test/VisualizationLoadingHOC.spec.tsx [hunk 2]
```diff
@@ -353,9 +353,10 @@ describe('VisualizationLoadingHOC', () => {
 
             return testUtils.delay().then(() => {
                 expect(pushData).toHaveBeenCalledWith({
-                    result: oneMeasureResponse
+                    result: oneMeasureResponse,
+                    properties: { sortItems: undefined }
```
Comment (lhorak) [inline src/components/core/base/test/VisualizationLoadingHOC.spec.tsx]: why is this needed?

File: src/components/filters/AttributeFilter/AttributeElements.tsx [hunk 3]
```diff
@@ -198,16 +200,25 @@ export class AttributeElements extends React.PureComponent<IAttributeElementsPro
...
+        let currentGetValidElementsPromise: Promise<any> = null;
+        // The promise needs to reset here because we are not setting it synchroneously
+        // and even this small delay is sometimes too late
+        this.getValidElementsPromise = null;
+
         uriPromise
             .then((uri: string) => {
                 const objectId = getObjectIdFromUri(uri);
-                return this.sdk.md.getValidElements(
+                this.getValidElementsPromise = currentGetValidElementsPromise = this.sdk.md.getValidElements(
```
Comment (lhorak) [inline src/components/filters/AttributeFilter/AttributeElements.tsx]: I know it's a shorthand, but I'd still assign it in multiple rows, as it's more clear to read

Comment (lhorak) [inline src/components/filters/AttributeFilter/AttributeElements.tsx]: I'd make the shorthand into separate assignments, it'd make it more readable

Comment (lhorak) [inline src/components/filters/AttributeFilter/AttributeElements.tsx]: I'd make the shorthand into separate assignments, it'd make it more readable

Comment (lhorak) [inline src/components/filters/AttributeFilter/AttributeElements.tsx]: I'd make the shorthand into separate assignments, it'd make it more readable

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1806/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1290/ : SUCCESS in 3m 29s

**Comment by lhorak** [comment]
LGTM

**Comment by lhorak** [comment]
LGTM

**Comment by lhorak** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/427/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1300/ : SUCCESS in 4m 07s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/752/ : SUCCESS in 17m 02s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/352/ : SUCCESS in 25m 42s
